RHOARDOC-1603: Operator-based launcher installation procedure

inoxx03 opened the pull request.

inoxx03 commented:
@cmoulliard Would you please look at the operator-based installation procedure and let me know whether it seems OK to you?
Since this pull request also contains some additional related changes, please feel free to look at the rendered version: https://deploy-preview-1274--launcher-documentation.netlify.com/docs/minishift-installation#installing-launcher-tool-using-an-operator_minishift

Thank you:)

@snelluli1 Please take a look at the procedure an let me know if you see something that should be changed :) Thank you.

cmoulliard commented on docs/topics/proc_installing-launcher-operator.adoc:
```diff
@@ -0,0 +1,112 @@
+* access role definitions for the launcher operator, and
+* access role bindings for the Launcher operator.
+
+. Create the custom resource for the Launcher application.
```
I would suggest to say `Install the following custom resource in order to initiate the deployment of the launcher. WDYT ? @inoxx03 

It should be great that Launcher's team document the Launcher Spec API - https://github.com/fabric8-launcher/launcher-operator/blob/master/pkg/apis/launcher/v1alpha2/launcher_types.go and that you add a remark here in order to explain the different parameters that we can customize within the CR and that they also release the operator

inoxx03 replied:
@cmoulliard Thank you. I have rephrased the instructions based on your suggestion.
I have also discussed changing the contents of the `deploy` directory to a resource list, but was unable to get the changes into the launcher-operator soon enough. Therofe I have replaced the clone step with a ZIP download step and filed a follow-up JIRA to create a resource list and subsequently update the instructions.  I am also filing a JIRA to document the Launcher spec API, which I will do post-summit:)

snelluli1 commented on docs/minishift-installation/master.adoc:
```diff
@@ -9,29 +9,30 @@ include::topics/templates/document-attributes.adoc[]
-This guide walks you through the process of installing the {name-launcher} tool to run on a local cloud as provisioned by a {OpenShiftLocal}.
-This includes link:https://www.openshift.org/minishift/[{Minishift}], an all-in-one VM with link:https://www.okd.io[OKD^], or link:https://developers.redhat.com/products/cdk/overview/[{CDK}], a VM that includes OpenShift Container Platform.
+This guide walks you through the process of installing the {name-launcher} tool on an {OpenShiftCluster}.
+To test the launcher tool locally on your machine
```
The link is commented out, and so is not visible in the rendered version. Please check.

inoxx03 replied:
Thank you for catching this. I can fix this, but I need to re-create the link, because I changed the original chapter:)

snelluli1 commented on docs/topics/proc_installing-launcher-operator.adoc:
```diff
@@ -0,0 +1,112 @@
+* a custom resource definition for the Launcher operator,
+* a service account for the Launcher operator,
+* access role definitions for the launcher operator, and
+* access role bindings for the Launcher operator.
```
We do not need the commas or the period at the end of the list items. The "and" at the end of the second last item can be removed too. So, the list would look like this:
Based on the templates, OpenShift automatically creates:

* a custom resource definition for the Launcher operator
* a service account for the Launcher operator
* access role definitions for the launcher operator
* access role bindings for the Launcher operator

snelluli1 commented on docs/topics/proc_installing-launcher-operator.adoc:
```diff
@@ -0,0 +1,112 @@
+.Prerequisites
+
+* Admin access to an OpenShift cluster.
+z
```
Please remove the "z".

inoxx03 replied:
done:)

snelluli1 commented on docs/topics/proc_starting-and-configuring-the-openshiftlocal-for-the-launcher-tool.adoc:
```diff
@@ -1,27 +1,27 @@
+This chapter contains instructions for starting the {Minishift} and configuring it to execute the {name-launcher} tool.
 
-NOTE: Starting your {OpenShiftLocal} can trigger a download of large virtual machines or Linux container images. This can take a long time. Subsequent startups are expected to be shorter as long as the virtual machines and Linux container images remain cached.
+NOTE: Starting your {Minishift} can trigger a download of large virtual machines or Linux container images. This can take a long time. Subsequent startups are expected to be shorter as long as the virtual machines and Linux container images remain cached.
```
as long as--> if

inoxx03 replied:
fixed:)

inoxx03 commented:
@snelluli1 Thak you for the comments, I have addressed them and updated the PR. Please re-review when you have a chance, and ack if everything seems OK to you. Please feel free to ping me if you have questions:) Thank you.
